### PR TITLE
Enables linting for webapp ts files. 

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

Leveraging "@typescript-eslint" and "@angular-eslint/eslint-plugin", enables linting for webapp ts files. 
Moves `bootstrapper` back into `js` folder, it was too ambitious for me to think we would convert that to typescript in this iteration. 
Fixes linting errors. 
Updates some unit tests to follow the same structure in terms of configuring testing module, stubbing and injecting. 

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by dianabarsan** [comment]
Please have a look @latin-panda . Thanks!

File: webapp/src/ts/services/message-contact.service.ts [hunk 1]
```diff
@@ -71,9 +71,9 @@ export class MessageContactService {
   }
 
   isRelevantChange(change, conversation?) {
-    return (change.doc && change.doc.kujua_message)
-      || (change.doc && change.doc.sms_message)
-      || change.deleted
-      || (conversation && conversation.messages && conversation.messages.find(message => message.doc._id === change.id));
+    return (change.doc && change.doc.kujua_message) ||
```
Comment (latin-panda) [inline webapp/src/ts/services/message-contact.service.ts:74]: 😅 Either way works, but usually I understand better when I read code and the operator is in front, like "prefixing" that the following lines are still part of the condition and not regular statements (like fn, assignments, etc). 

Is this breaking a lint rule?

**Review by latin-panda — APPROVED** [review]
So happy that we have lint again! Thanks!
Just left a couple of comments but not really important, this is good :)

File: webapp/src/ts/services/message-contact.service.ts (hunk 1 above)
Comment (dianabarsan) [reply webapp/src/ts/services/message-contact.service.ts:74]: I like them better at the and and, yea it was breaking the line length rule with the operator at the end :(

Comment (dianabarsan) [reply webapp/src/ts/services/message-contact.service.ts:74]: The last line is shortened by 3 chars if the operator is on the previous line.

File: webapp/src/ts/services/geolocation.service.ts [hunk 2]
```diff
@@ -79,7 +79,8 @@ export class GeolocationService {
         message: 'Geolocation API unavailable.',
       };
     } else {
-      this.watcher = window.navigator.geolocation.watchPosition(this.success.bind(this), this.failure.bind(this), this.GEO_OPTIONS);
+      this.watcher =
+        window.navigator.geolocation.watchPosition(this.success.bind(this), this.failure.bind(this), this.GEO_OPTIONS);
```
[the remark this answers is not in the document]
Comment (dianabarsan) [reply webapp/src/ts/services/geolocation.service.ts]: changed, thanks!

File: webapp/src/ts/providers/parse.provider.ts [hunk 3]
```diff
@@ -325,23 +376,24 @@ class ASTCompiler {
...
+    return args.join(',');
   }
 
   addReturnStmt(result) {
     this.stmts.push(`return ${result};`);
   }
 
   cleanup() {
-    this.ast = this.cAst = this.stmts = this.cStmts = this.declarations = this.pipes = this.pipeNameVsIsPureMap = undefined;
+    this.ast = this.cAst = this.stmts = this.cStmts = this.declarations = this.pipes = this.pipeNameVsIsPureMap =
```
[the remark this answers is not in the document]
Comment (dianabarsan) [reply webapp/src/ts/providers/parse.provider.ts]: Changed!

**Pull request merged**